tests/components/yamaha/test_media_player.py

```diff
@@ -23,54 +26,142 @@ def __init__(self, ctrl_url, name, zones=None):
+    with patch("rxv.find", return_value=[device]):
+        assert await async_setup_component(
+            hass, mp.DOMAIN, {"media_player": {"platform": "yamaha"}}
         )
+        await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is not None
+    assert state.state == "off"
+
+
+async def test_setup_discovery(hass, device, main_zone):
+    """Test set up integration via discovery."""
+    discovery_info = {
+        "name": "Yamaha Receiver",
+        "model_name": "Yamaha",
+        "control_url": "http://receiver",
+        "description_url": "http://receiver/description",
+    }
+    await async_load_platform(
+        hass, mp.DOMAIN, "yamaha", discovery_info, {mp.DOMAIN: {}}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is not None
+    assert state.state == "off"
+
+
+async def test_setup_zone_ignore(hass, device, main_zone):
+    """Test set up integration without host."""
+    assert await async_setup_component(
+        hass,
+        mp.DOMAIN,
+        {
+            "media_player": {
+                "platform": "yamaha",
+                "host": "127.0.0.1",
+                "zone_ignore": "Main zone",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is None
+
+
+async def test_enable_output(hass, device, main_zone):
+    """Test enable output service."""
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+
+    port = "hdmi1"
+    enabled = True
+    data = {
+        "entity_id": "media_player.yamaha_receiver_main_zone",
+        "port": port,
+        "enabled": enabled,
+    }
+
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_ENABLE_OUTPUT, data, True)
+
+    assert main_zone.enable_output.call_count == 1
+    assert main_zone.enable_output.call_args == call(port, enabled)
+
+
+async def test_select_scene(hass, device, main_zone, caplog):
+    """Test select scene service."""
+    scene_prop = PropertyMock(return_value=None)
+    type(main_zone).scene = scene_prop
+
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+
+    scene = "TV Viewing"
+    data = {
+        "entity_id": "media_player.yamaha_receiver_main_zone",
+        "scene": scene,
+    }
 
-    def tearDown(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_SELECT_SCENE, data, True)
 
-    def enable_output(self, port, enabled):
-        """Enable output on a specific port."""
-        data = {
-            "entity_id": "media_player.yamaha_receiver_main_zone",
-            "port": port,
-            "enabled": enabled,
-        }
+    assert scene_prop.call_count == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5662368)